freelist.go

```diff
@@ -171,7 +172,9 @@ func (f *freelist) read(p *page) {
 	copy(f.ids, ids)
 
 	// Make sure they're sorted.
-	sort.Sort(pgids(f.ids))
+	if !pgids(f.ids).isSorted() {
```

Comment:
Yes. How does sort.Sort behave on already sorted data? Do you have benchmarks to help understand how much of a win this is?